src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -233,9 +233,9 @@ def update_data(self):
             if not self.pubsub[i]:
                 pattern = self.TRANSCEIVER_KEY_PATTERN
                 self.pubsub[i] = mibs.get_redis_pubsub(self.statedb[i], self.statedb[i].STATE_DB, pattern)
```

Comment:
Move to `__init__()` ?

Reply:
This function is currently in sonic_ax_impl/mibs/__init__.py

Reply:
Sorry, I mean the section is only needed once after each time db connected.

```
        for i in range(len(self.statedb)):
            if not self.pubsub[i]:
                pattern = self.TRANSCEIVER_KEY_PATTERN
                self.pubsub[i] = mibs.get_redis_pubsub(self.statedb[i], self.statedb[i].STATE_DB, pattern)
```

---
In reply to: [451174442](https://github.com/Azure/sonic-snmpagent/pull/137#discussion_r451174442) [](ancestors = 451174442)